src/Hyperion/Extensions/TypeEx.cs

```diff
@@ -125,6 +125,14 @@ private static Type GetTypeFromManifestName(Stream stream, DeserializerSession s
             return TypeNameLookup.GetOrAdd(byteArr, b =>
             {
                 var shortName = StringEx.FromUtf8Bytes(b.Bytes, 0, b.Bytes.Length);
+                if (shortName.Contains("System.Private.CoreLib,%core%") && CoreAssemblyQualifiedName.Contains("mscorlib, Version="))
```

Comment:
Isn't this if/else statement something that could be replaced by compilation flags (iike `#if NETCORE`)? I'm not sure if this check is something that we need to call at runtime for potentially every single type when we know that `CoreAssemblyQualifiedName` will never change in the application lifetime.

Reply:
It isn't my commit but I'm not sure if we don't need runtime checks. Because the whole purpose of this PR is to provide interoperability between the .NET Framework and .NET Core, thus we might need runtime checks in case serialize/deserialize between those two frameworks. As far as I understand Hyperion embeds the namespace into the serialized payload for types.

I'll check it and update the code if it's necessary. Thanks for pointing out.

Reply:
`
if (shortName.Contains("System.Private.CoreLib,%core%") && CoreAssemblyQualifiedName.Contains("mscorlib, Version="))`

The second predication can be replaced by etc ```#if NET461```  
But i more like define a global variable
`private static readonly string IsFullNetFramework = CoreAssemblyQualifiedName.Contains("mscorlib, Version="))`

`if (shortName.Contains("System.Private.CoreLib,%core%") && IsFullNetFramework`

Reply:
The reason here is that we want to avoid unnecessary branches and operations. While it's quite possible that .NET Core 3.0 knows how to optimize static readonly checks, it's wasteful to defer checks, which result is known at compile time, back to the runtime for every type call. It's only a microoptimization, but it's also a sane practice to do.

A middle ground here is to make `IsFullNetFramework` a const defined in #if/else pragmas and put it as a first segment of && expression - since consts cannot be changed by reflection (unlike readonly values), they are always optimized on every .NET platform.

Reply:
Fixed In Blind-Striker#1

`#if NETSTANDARD`
`#if NET45`

Reply:
Ok, everything seems to be in order. I merged @humhei 's PR and pull the latest CI improvements from the dev branch. All tests are passed. 

After @akselarzuman's PR #119 merged I need to update test files (mentioned in https://github.com/akkadotnet/Hyperion/pull/116#issue-282801606) for .NET Core 3.0 and run the tests again.